docs/src/main/paradox/proto.md

```diff
@@ -0,0 +1,40 @@
+# Protobuf file format
+
+gRPC uses the Protobuf `.proto` file format to define your messages, services
+and some aspects of the code generation.
+
+For an in-depth description see the [official documentation](https://developers.google.com/protocol-buffers/docs/proto),
+but here are a few quick examples:
+
+## Messages
+
+Messages are defined in the same way protobuf definitions are used for serialization:
+
+@@snip [helloworld.proto]($root$/../plugin-tester-scala/src/main/protobuf/helloworld.proto) { #messages }
+
+## Services
+
+Those messages can be used to define services:
+
+@@snip [helloworld.proto]($root$/../plugin-tester-scala/src/main/protobuf/helloworld.proto) { #services }
+
+Both the request and the response consists of either a single message or a stream of messages.
```

Comment:
Something I wondered as a newbie was, how can I make a Unit/void function, or one with no parameters. Maybe add that it is not possible to do that here? (Unless I'm wrong ofc)

Reply:
Good one: I think the convention is to use the 'empty' object that Google provides for this (https://github.com/google/protobuf/blob/master/src/google/protobuf/empty.proto), but it might take a small section (also documenting how to generate code for proto's from dependencies). I'll try to work it in.